patches/chromium/enable_reset_aspect_ratio.patch

```diff
@@ -0,0 +1,37 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Cheng Zhao <zcbenz@gmail.com>
+Date: Thu, 4 Oct 2018 14:57:02 -0700
+Subject: feat: enable setting aspect ratio to 0
+
+Make SetAspectRatio accept 0 as valid input, which would reset to null.
+
+diff --git a/ui/views/widget/desktop_aura/desktop_window_tree_host_win.cc b/ui/views/widget/desktop_aura/desktop_window_tree_host_win.cc
+index 3b132d4d7618e11b851b315e82fa67442c852fb3..6cee855828fc8069b7cd7d429dcda5dc9bd7ecf8 100644
+--- a/ui/views/widget/desktop_aura/desktop_window_tree_host_win.cc
++++ b/ui/views/widget/desktop_aura/desktop_window_tree_host_win.cc
+@@ -484,7 +484,6 @@ void DesktopWindowTreeHostWin::SetOpacity(float opacity) {
+ }
+ 
+ void DesktopWindowTreeHostWin::SetAspectRatio(const gfx::SizeF& aspect_ratio) {
+-  DCHECK(!aspect_ratio.IsEmpty());
```

Comment:
@zcbenz why are we removing this DCHECK?  if aspect_ratio.IsEmpty() then the next line won't work will it?

Reply:
`aspect_ratio.IsEmpty()` could also be true when it is something like `(0, 100)`.

Reply:
@zcbenz true, but i think this will crash on divide-by-zero if we don't at least ensure that height > 0, won't it?

Reply:
Yeah it will crash if height == 0, but I think it should be fine crashing here? If someone pass a 0 height then the code will crash anyway with or without the DCHECK. The purpose of the DCHECK is prevent passing 0 to `SetAspectRatio`, instead of prevent the divide-by-zero crash.

Also we won't trigger the divide-by-zero crash in Electron since we are always passing `(ratio, 100)` to this API.

Reply:
@zcbenz I guess to ask the question another way, it is necessary to remove this DCHECK?  What problem does it solve to do so?

Reply:
The Electron `win.setAspectRatio` API accepts `0` to cancel the aspect ratio, which internally passes `(0, 100)` to `DesktopWindowTreeHostWin::SetAspectRatio`, and it would cause an assertion with this DCHECK since `(0, 100)` is considered as an empty size.

Reply:
@zcbenz couldn't we change the DCHECK to 
`DCHECK(!aspect_ratio.height());` to avoid division by zero?

Reply:
I think i'd prefer that as well, since it's a clearer error

Reply:
@jkleinsc @codebytere Sure, I have changed the DCHECK to validate the height.